src/go/tmpl/templates/windows_startup.tmpl

```diff
@@ -88,6 +88,13 @@ $wmi[{{ $idx }}].SetGateways('{{ $iface.Gateway }}', 1) | Out-Null
 $wmi.SetGateways('{{ $iface.Gateway }}', 1) | Out-Null
         {{ end }}
     {{ end }}
+    {{ if $iface.DNS }}
+        {{ if gt $length 1 }}
+$wmi[{{ $idx }}].SetDNSServerSearchOrder(@('{{ stringsJoin $iface.DNS "', '" }}')) | Out-Null
+        {{ else }}
+$wmi.SetDNSServerSearchOrder(@('{{ stringsJoin $iface.DNS "', '" }}')) | Out-Null
+        {{ end }}
+    {{ end }}
```

Comment:
We also set the DNS server address below if this Windows VM is being added to a domain controller. How should we handle the case where DNS servers are provided for an interface, and the VM is being added to a domain controller, and the DNS servers provided are not the domain controller?

In order for AD to work across subnets, the interface that can reach the domain controller must use the domain controller as its DNS server. So, I'm thinking this might work as written, and if this VM is being added to a domain controller then its DNS server entry will just get overwritten below. We should probably document this procedure though, so no one gets confused if the DNS server they enter doesn't match.

Reply:
@glattercj any thoughts on this, or should we just leave it as-is?

Reply:
I think it's fine. Was going to test but don't have a DC topo handy. The main use case for this was external access for a VM, so folks (hopefully) know what they're doing if they also have configured a DC in the scenario config.